cmd/tools/rpcwrappers/main.go

```diff
@@ -0,0 +1,475 @@
+package {{.ServiceName}}
+
+import (
+	"context"
+
+	"{{.ServicePackagePath}}"
+	"google.golang.org/grpc"
+
+	"go.temporal.io/server/common/metrics"
+)
+`)
+
+	writeTemplatedMethods(w, service, "metricsClient", `
+func (c *metricClient) {{.Method}}(
+	ctx context.Context,
+	request {{.RequestType}},
+	opts ...grpc.CallOption,
+) (_ {{.ResponseType}}, retError error) {
+
+	scope, stopwatch := c.startMetricsRecording(metrics.{{.MetricPrefix}}{{.Method}}Scope)
+	defer func() {
+		c.finishMetricsRecording(scope, stopwatch, retError)
+	}()
+
+	return c.client.{{.Method}}(ctx, request, opts...)
+}
+`)
+	// TODO: some history methods did not touch metrics. should we preserve this?
+	// DescribeHistoryHost
+	// RemoveTask
+	// CloseShard
+	// GetShard
+	// RebuildMutableState
+	// DescribeMutableState
+	// TODO: DeleteWorkflowExecution didn't work like the others in history (the code looked
+	// like the frontend/admin client version). should we preserve that?
+}
+
+func generateRetryableClient(w io.Writer, service service) {
+	writeTemplatedCode(w, service, `
+package {{.ServiceName}}
+
+import (
+	"context"
+
+	"{{.ServicePackagePath}}"
+	"google.golang.org/grpc"
+
+	"go.temporal.io/server/common/backoff"
+)
+`)
+
+	writeTemplatedMethods(w, service, "retryableClient", `
+func (c *retryableClient) {{.Method}}(
+	ctx context.Context,
+	request {{.RequestType}},
+	opts ...grpc.CallOption,
+) ({{.ResponseType}}, error) {
+	var resp {{.ResponseType}}
+	op := func() error {
+		var err error
+		resp, err = c.client.{{.Method}}(ctx, request, opts...)
+		return err
+	}
+	err := backoff.Retry(op, c.policy, c.isRetryable)
+	return resp, err
+}
+`)
+}
+
+func callWithFile(f func(io.Writer, service), service service, filename string, licenseText string) {
+	w, err := os.Create(filename + "_gen.go")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Fprintf(w, "%s\n// Code generated by cmd/tools/rpcwrappers. DO NOT EDIT.\n", licenseText)
+	f(w, service)
+	err = w.Close()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func readLicenseFile(path string) string {
+	text, err := os.ReadFile(path)
+	if err != nil {
+		panic(err)
+	}
+	var lines []string
+	for _, line := range strings.Split(string(text), "\n") {
+		lines = append(lines, strings.TrimRight("// "+line, " "))
+	}
+	return strings.Join(lines, "\n") + "\n"
+}
+
+func main() {
+	serviceFlag := flag.String("service", "", "which service to generate rpc client wrappers for")
+	licenseFlag := flag.String("licence_file", "../../LICENSE", "path to license to copy into header")
+	flag.Parse()
+
+	i := slices.IndexFunc(services, func(s service) bool { return s.name == *serviceFlag })
+	if i < 0 {
+		panic("unknown service")
+	}
+	svc := services[i]
+
+	licenseText := readLicenseFile(*licenseFlag)
+
+	callWithFile(svc.clientGenerator, svc, "client", licenseText)
+	callWithFile(svc.metricGenerator, svc, "metricClient", licenseText)
+	callWithFile(svc.retryableGenerator, svc, "retryableClient", licenseText)
```

Comment:
Let's follow go best practices for all new file names:
```suggestion
	callWithFile(svc.metricGenerator, svc, "metric_client", licenseText)
	callWithFile(svc.retryableGenerator, svc, "retryable_client", licenseText)
```

Reply:
yes, please rename other too.

Actually I believe we need to rename all our files to the snake_case.